<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Convenient functions for mapping orthologous genes using ENSEMBL BioMART â€¢ orthologsBioMART</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Convenient functions for mapping orthologous genes using ENSEMBL BioMART">
<meta property="og:description" content="Wraps around biomaRt package to provide convenient functions for mapping orthologous genes using ENSEMBL BioMART">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">orthologsBioMART</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">

<div id="convenient-functions-for-mapping-orthologous-genes-using-ensembl-biomart" class="section level2">
<h2 class="hasAnchor">
<a href="#convenient-functions-for-mapping-orthologous-genes-using-ensembl-biomart" class="anchor"></a>Convenient functions for mapping orthologous genes using ENSEMBL BioMART</h2>
<p>The package wraps around biomaRt package to provide a fast way of ortholog mapping.</p>
<p><a href="https://zenodo.org/badge/latestdoi/240325005"><img src="https://zenodo.org/badge/240325005.svg" alt="DOI"></a></p>
</div>
<div id="installation-r" class="section level2">
<h2 class="hasAnchor">
<a href="#installation-r" class="anchor"></a>Installation R</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span>(<span class="st">"BiocManager"</span>)
BiocManager<span class="op">::</span><span class="kw"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"vitkl/orthologsBioMART"</span>), <span class="dt">dependencies=</span>T)</code></pre></div>
</div>
<div id="usage-r" class="section level2">
<h2 class="hasAnchor">
<a href="#usage-r" class="anchor"></a>Usage R</h2>
<p>Mapping between any species:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">findOrthologs =<span class="st"> </span><span class="cf">function</span>(<span class="dt">datasets_FROM_TO =</span> <span class="kw">loadBIOMARTdatasets</span>(<span class="dt">from =</span> <span class="st">"hsapiens_gene_ensembl"</span>,
                                                                <span class="dt">to =</span> <span class="st">"mmusculus_gene_ensembl"</span>),
                         <span class="dt">from_filters =</span> <span class="st">"hgnc_symbol"</span>, <span class="co"># type of input identifier</span>
                         <span class="dt">from_values =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"TP53"</span>, <span class="st">"TERT"</span>), <span class="co"># gene identifiers to map</span>
                         <span class="dt">to_attributes =</span> <span class="st">"external_gene_name"</span>, <span class="co"># type of output identifier</span>
                         <span class="co"># column in the input database that maps to ENSEMBL id of the output database:</span>
                         <span class="dt">to_homolog_attribute =</span> <span class="st">"mmusculus_homolog_ensembl_gene"</span>,  
                         <span class="co"># column with ENSEMBL gene id in the input database:  </span>
                         <span class="dt">from_gene_id_name =</span> <span class="st">"human_ensembl_gene_id"</span>, 
                         <span class="co"># column with ENSEMBL gene id in the output database:  </span>
                         <span class="dt">to_gene_id_name =</span> <span class="st">"mouse_ensembl_gene_id"</span>)</code></pre></div>
<p>Mapping human to mouse and mouse to human:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">findOrthologsHsMm</span>(<span class="dt">from_filters =</span> <span class="st">"hgnc_symbol"</span>,
  <span class="dt">from_values =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"TP53"</span>,<span class="st">"TERT"</span>), 
  <span class="dt">to_attributes =</span> <span class="st">"external_gene_name"</span>)

<span class="kw">findOrthologsMmHs</span>(<span class="dt">from_filters =</span> <span class="st">"ensembl_gene_id"</span>,
  <span class="dt">from_values =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"ENSMUSG00000059552"</span>, <span class="st">"ENSMUSG00000021611"</span>),
  <span class="dt">to_attributes =</span> <span class="st">"hgnc_symbol"</span>)</code></pre></div>
<p>To find the correct <code>to_homolog_attribute</code> between species use:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">attr =<span class="st"> </span><span class="kw">attributesFiltersFromTo</span>(<span class="dt">datasets_FROM_TO =</span> <span class="kw">loadBIOMARTdatasets</span>())

<span class="co"># examine attr</span>
attr

<span class="co"># find the desired column name</span>
attr<span class="op">$</span>from_attributes[<span class="kw"><a href="https://rdrr.io/r/base/grep.html">grep</a></span>(<span class="st">"Mouse gene stable ID"</span>, attr, <span class="dt">ignore.case =</span> T),]</code></pre></div>
</div>
<div id="installation-python" class="section level2">
<h2 class="hasAnchor">
<a href="#installation-python" class="anchor"></a>Installation python</h2>
<div class="sourceCode"><pre class="sourceCode python"><code class="sourceCode python">pip install git<span class="op">+</span>https:<span class="op">//</span>github.com<span class="op">/</span>vitkl<span class="op">/</span>orthologsBioMART.git</code></pre></div>
</div>
<div id="usage-python" class="section level2">
<h2 class="hasAnchor">
<a href="#usage-python" class="anchor"></a>Usage python</h2>
<p>Mapping between any species:</p>
<div class="sourceCode"><pre class="sourceCode python"><code class="sourceCode python"><span class="co"># find correct dataset name for your species</span>
<span class="im">from</span> pybiomart <span class="im">import</span> Server
server <span class="op">=</span> Server(host<span class="op">=</span><span class="st">'http://www.ensembl.org'</span>)
server.marts[<span class="st">'ENSEMBL_MART_ENSEMBL'</span>].list_datasets()

<span class="im">from</span> <span class="im">import</span> FindOrthologs 
<span class="co"># then create the find orthogues object using correct datasets and attributes</span>
<span class="co"># use help(FindOrthologs)</span>
hs2mm <span class="op">=</span> FindOrthologs(
          host <span class="op">=</span> <span class="st">'http://www.ensembl.org'</span>,
          mart <span class="op">=</span> <span class="st">'ENSEMBL_MART_ENSEMBL'</span>,
          from_dataset <span class="op">=</span> <span class="st">'hsapiens_gene_ensembl'</span>,
          to_dataset <span class="op">=</span> <span class="st">'mmusculus_gene_ensembl'</span>,
          from_filters <span class="op">=</span> <span class="st">'hgnc_symbol'</span>,
          from_values <span class="op">=</span> [<span class="st">'TP53'</span>, <span class="st">'TERT'</span>],
          to_attributes <span class="op">=</span> <span class="st">'external_gene_name'</span>,
          to_homolog_attribute <span class="op">=</span> <span class="st">'mmusculus_homolog_ensembl_gene'</span>,
          from_gene_id_name <span class="op">=</span> <span class="st">'human_ensembl_gene_id'</span>,
          to_gene_id_name <span class="op">=</span> <span class="st">'mouse_ensembl_gene_id'</span>
    )
    
hs2mm.<span class="bu">map</span>()</code></pre></div>
<p>Mapping mouse to human and vice-versa:</p>
<div class="sourceCode"><pre class="sourceCode python"><code class="sourceCode python">findOrthologsMmHs(from_filters <span class="op">=</span> <span class="st">'link_ensembl_gene_id'</span>,
                  from_values <span class="op">=</span> [<span class="st">'ENSMUSG00000059552'</span>, <span class="st">'ENSMUSG00000021611'</span>]).<span class="bu">map</span>()
findOrthologsMmHs(from_filters <span class="op">=</span> <span class="st">'external_gene_name'</span>,
                  from_values <span class="op">=</span> [<span class="st">'Trp53'</span>, <span class="st">'Tert'</span>]).<span class="bu">map</span>()

findOrthologsHsMm(from_filters <span class="op">=</span> <span class="st">'link_ensembl_gene_id'</span>,
                  from_values <span class="op">=</span> [<span class="st">'ENSG00000141510'</span>, <span class="st">'ENSG00000164362'</span>]).<span class="bu">map</span>()
findOrthologsHsMm(from_filters <span class="op">=</span> <span class="st">'hgnc_symbol'</span>,
                  from_values <span class="op">=</span> [<span class="st">'TP53'</span>, <span class="st">'TERT'</span>]).<span class="bu">map</span>()</code></pre></div>
</div>

  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <div class="license">
<h2>License</h2>
<ul class="list-unstyled">
<li>Apache 2.0</li>
</ul>
</div>
<div class="developers">
<h2>Developers</h2>
<ul class="list-unstyled">
<li>Vitalii Kleshchevnikov <br><small class="roles"> Maintainer </small>  </li>
</ul>
</div>

  </div>
</div>


      <footer><div class="copyright">
  <p>Developed by Vitalii Kleshchevnikov.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
