<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>map orthologous genes using ENSEMBL BioMART — findOrthologs • orthologsBioMART</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="map orthologous genes using ENSEMBL BioMART — findOrthologs" />
<meta property="og:description" content="findOrthologs(): map orthologous genes using ENSEMBL BioMART (getBM) from one species to another species. This function retrieves ids by 500 at a time to decrease the load on ENSEMBL servers.
findOrthologsHsMm(): map orthologous genes using ENSEMBL BioMART from human to mouse
findOrthologsMmHs(): map orthologous genes using ENSEMBL BioMART from mouse to human
loadBIOMARTdatasets(): Load 2 ENSEMBL BioMART datasets for finding orthologs. Details useMart
attributesFiltersFromTo(): List filters and attributes available in each dataset. Details: listAttributes" />
<meta name="twitter:card" content="summary" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">orthologsBioMART</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>map orthologous genes using ENSEMBL BioMART</h1>
    
    <div class="hidden name"><code>findOrthologs.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>findOrthologs(): map orthologous genes using ENSEMBL BioMART (<a href='https://rdrr.io/pkg/biomaRt/man/getBM.html'>getBM</a>) from one species to another species. This function retrieves ids by 500 at a time to decrease the load on ENSEMBL servers.</p>
<p>findOrthologsHsMm(): map orthologous genes using ENSEMBL BioMART from human to mouse</p>
<p>findOrthologsMmHs(): map orthologous genes using ENSEMBL BioMART from mouse to human</p>
<p>loadBIOMARTdatasets(): Load 2 ENSEMBL BioMART datasets for finding orthologs. Details <a href='https://rdrr.io/pkg/biomaRt/man/useMart.html'>useMart</a></p>
<p>attributesFiltersFromTo(): List filters and attributes available in each dataset. Details: <a href='https://rdrr.io/pkg/biomaRt/man/listAttributes.html'>listAttributes</a></p>
    </div>

    <pre class="usage"><span class='fu'>findOrthologs</span>(<span class='kw'>datasets_FROM_TO</span> <span class='kw'>=</span> <span class='fu'>loadBIOMARTdatasets</span>(<span class='kw'>from</span> <span class='kw'>=</span>
  <span class='st'>"hsapiens_gene_ensembl"</span>, <span class='kw'>to</span> <span class='kw'>=</span> <span class='st'>"mmusculus_gene_ensembl"</span>),
  <span class='kw'>from_filters</span> <span class='kw'>=</span> <span class='st'>"hgnc_symbol"</span>, <span class='kw'>from_values</span> <span class='kw'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='st'>"TP53"</span>, <span class='st'>"TERT"</span>),
  <span class='kw'>to_attributes</span> <span class='kw'>=</span> <span class='st'>"external_gene_name"</span>,
  <span class='kw'>to_homolog_attribute</span> <span class='kw'>=</span> <span class='st'>"mmusculus_homolog_ensembl_gene"</span>,
  <span class='kw'>from_gene_id_name</span> <span class='kw'>=</span> <span class='st'>"human_ensembl_gene_id"</span>,
  <span class='kw'>to_gene_id_name</span> <span class='kw'>=</span> <span class='st'>"mouse_ensembl_gene_id"</span>)

<span class='fu'>findOrthologsHsMm</span>(<span class='kw'>from_filters</span> <span class='kw'>=</span> <span class='st'>"hgnc_symbol"</span>, <span class='kw'>from_values</span> <span class='kw'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='st'>"TP53"</span>,
  <span class='st'>"TERT"</span>), <span class='kw'>to_attributes</span> <span class='kw'>=</span> <span class='st'>"external_gene_name"</span>)

<span class='fu'>findOrthologsMmHs</span>(<span class='kw'>from_filters</span> <span class='kw'>=</span> <span class='st'>"ensembl_gene_id"</span>,
  <span class='kw'>from_values</span> <span class='kw'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='st'>"ENSMUSG00000059552"</span>, <span class='st'>"ENSMUSG00000021611"</span>),
  <span class='kw'>to_attributes</span> <span class='kw'>=</span> <span class='st'>"hgnc_symbol"</span>)

<span class='fu'>loadBIOMARTdatasets</span>(<span class='kw'>from</span> <span class='kw'>=</span> <span class='st'>"hsapiens_gene_ensembl"</span>,
  <span class='kw'>to</span> <span class='kw'>=</span> <span class='st'>"mmusculus_gene_ensembl"</span>)

<span class='fu'>attributesFiltersFromTo</span>(<span class='kw'>datasets_FROM_TO</span> <span class='kw'>=</span> <span class='fu'>loadBIOMARTdatasets</span>())</pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>datasets_FROM_TO</th>
      <td><p>environment containing "Mart" objects produced by <code>loadBIOMARTdatasets()</code>. It is convenient to specify datasets here rather than pre-loading. Full list of datasets in ENSEMBL BioMART: listDatasets("ensembl")</p></td>
    </tr>
    <tr>
      <th>from_filters</th>
      <td><p>Filters (one or more) that should be used in the query in the "from" species database. A possible list of filters can be retrieved using <code>attributesFiltersFromTo(datasets_FROM_TO = loadBIOMARTdatasets())</code>.</p></td>
    </tr>
    <tr>
      <th>from_values</th>
      <td><p>Values of the filter, e.g. vector of affy IDs in the "from" species database. If multiple filters are specified then the argument should be a list of vectors of which the position of each vector corresponds to the position of the filters in the filters argument. Please limit the number of values to less than 500.</p></td>
    </tr>
    <tr>
      <th>to_attributes</th>
      <td><p>Attributes you want to retrieve from the "to" species database. A possible list of attributes can be retrieved using the function <code>attributesFiltersFromTo(datasets_FROM_TO = loadBIOMARTdatasets())</code>.</p></td>
    </tr>
    <tr>
      <th>to_homolog_attribute</th>
      <td><p>Attribute that returns homolog gene id in the "from" database</p></td>
    </tr>
    <tr>
      <th>from_gene_id_name</th>
      <td><p>name of the column containing the ENSEMBL gene ids of "from" species</p></td>
    </tr>
    <tr>
      <th>to_gene_id_name</th>
      <td><p>name of the column containing the ENSEMBL gene ids of "from" species</p></td>
    </tr>
    <tr>
      <th>from</th>
      <td><p>ENSEMBL biomart dataset for a species whose identifiers you want to convert. Full list of possible options: listDatasets("ensembl")</p></td>
    </tr>
    <tr>
      <th>to</th>
      <td><p>ENSEMBL biomart dataset for a species whose identifiers you want convert into.</p></td>
    </tr>
    <tr>
      <th>datasets_FROM_TO</th>
      <td><p>environment containing "Mart" objects produced by <code>loadBIOMARTdatasets()</code></p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>findOrthologs(): data.frame containing the mapping of homologous genes from one species to another species including <code>to_attributes</code></p>
<p>findOrthologsHsMm(): data.frame containing the mapping of homologous genes from human to mouse including <code>to_attributes</code></p>
<p>findOrthologsMmHs(): data.frame containing the mapping of homologous genes from mouse to human including <code>to_attributes</code></p>
<p>loadBIOMARTdatasets(): environment containing 2 object of class 'Mart', one for each species dataset</p>
<p>attributesFiltersFromTo(): list containing 4 vector of either filters or attributes for each dataset</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='fu'>findOrthologs</span>()</div><div class='output co'>#&gt;   human_ensembl_gene_id mouse_ensembl_gene_id external_gene_name hgnc_symbol
#&gt; 1       ENSG00000141510    ENSMUSG00000059552              Trp53        TP53
#&gt; 2       ENSG00000164362    ENSMUSG00000021611               Tert        TERT</div><div class='input'><span class='fu'>findOrthologsHsMm</span>()</div><div class='output co'>#&gt;   human_ensembl_gene_id mouse_ensembl_gene_id external_gene_name hgnc_symbol
#&gt; 1       ENSG00000141510    ENSMUSG00000059552              Trp53        TP53
#&gt; 2       ENSG00000164362    ENSMUSG00000021611               Tert        TERT</div><div class='input'><span class='fu'>findOrthologsMmHs</span>()</div><div class='output co'>#&gt;   mouse_ensembl_gene_id human_ensembl_gene_id hgnc_symbol
#&gt; 1    ENSMUSG00000021611       ENSG00000164362        TERT
#&gt; 2    ENSMUSG00000059552       ENSG00000141510        TP53</div><div class='input'><span class='no'>attributes</span> <span class='kw'>=</span> <span class='fu'>attributesFiltersFromTo</span>(<span class='kw'>datasets_FROM_TO</span> <span class='kw'>=</span> <span class='fu'>loadBIOMARTdatasets</span>())
<span class='fu'><a href='https://rdrr.io/r/base/grep.html'>grep</a></span>(<span class='st'>"Mouse gene stable ID"</span>, <span class='no'>attributes</span>$<span class='no'>from_attributes</span>$<span class='no'>description</span>, <span class='kw'>ignore.case</span> <span class='kw'>=</span> <span class='no'>T</span>, <span class='kw'>value</span> <span class='kw'>=</span> <span class='no'>T</span>)</div><div class='output co'>#&gt; [1] "Algerian mouse gene stable ID"              
#&gt; [2] "Mouse gene stable ID"                       
#&gt; [3] "Northern American deer mouse gene stable ID"
#&gt; [4] "Ryukyu mouse gene stable ID"                
#&gt; [5] "Shrew mouse gene stable ID"                 
#&gt; [6] "Steppe mouse gene stable ID"                </div><div class='input'><span class='no'>attributes</span>$<span class='no'>from_attributes</span>[<span class='fu'><a href='https://rdrr.io/r/base/which.html'>which</a></span>(<span class='no'>attributes</span>$<span class='no'>from_attributes</span>$<span class='no'>description</span> <span class='kw'>==</span> <span class='st'>"Mouse gene name"</span>),]</div><div class='output co'>#&gt;                                        name     description     page
#&gt; 2213 mmusculus_homolog_associated_gene_name Mouse gene name homologs</div><div class='input'><span class='fu'><a href='https://rdrr.io/r/base/grep.html'>grep</a></span>(<span class='st'>"hgnc_symbol"</span>, <span class='no'>attributes</span>$<span class='no'>from_attributes</span>$<span class='no'>name</span>, <span class='kw'>value</span> <span class='kw'>=</span> <span class='no'>T</span>)</div><div class='output co'>#&gt; [1] "hgnc_symbol"</div><div class='input'><span class='co'># find the desired column name</span>
<span class='no'>attributes</span>$<span class='no'>from_attributes</span>[<span class='fu'><a href='https://rdrr.io/r/base/grep.html'>grep</a></span>(<span class='st'>"Mouse gene stable ID"</span>, <span class='no'>attributes</span>$<span class='no'>from_attributes</span>$<span class='no'>description</span>, <span class='kw'>ignore.case</span> <span class='kw'>=</span> <span class='no'>T</span>),]</div><div class='output co'>#&gt;                                  name
#&gt; 292     mspretus_homolog_ensembl_gene
#&gt; 2212   mmusculus_homolog_ensembl_gene
#&gt; 2308   pmbairdii_homolog_ensembl_gene
#&gt; 2692     mcaroli_homolog_ensembl_gene
#&gt; 2804     mpahari_homolog_ensembl_gene
#&gt; 2900 mspicilegus_homolog_ensembl_gene
#&gt;                                      description     page
#&gt; 292                Algerian mouse gene stable ID homologs
#&gt; 2212                        Mouse gene stable ID homologs
#&gt; 2308 Northern American deer mouse gene stable ID homologs
#&gt; 2692                 Ryukyu mouse gene stable ID homologs
#&gt; 2804                  Shrew mouse gene stable ID homologs
#&gt; 2900                 Steppe mouse gene stable ID homologs</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      <li><a href="#value">Value</a></li>
      <li><a href="#examples">Examples</a></li>
    </ul>

    <h2>Author</h2>
    <p>Vitalii Kleshchevnikov</p>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Vitalii Kleshchevnikov.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


